src/sbi/mod.rs

```diff
@@ -70,6 +70,7 @@ impl SbiMessage {
             sbi_spec::legacy::LEGACY_CONSOLE_PUTCHAR => Ok(SbiMessage::PutChar(args[0])),
             sbi_spec::legacy::LEGACY_CONSOLE_GETCHAR => Ok(SbiMessage::GetChar),
             sbi_spec::legacy::LEGACY_SET_TIMER => Ok(SbiMessage::SetTimer(args[0])),
+            sbi_spec::legacy::LEGACY_SHUTDOWN => Ok(SbiMessage::Reset(ResetFunction::shutdown())),

```

Comment:
Is is proper to call shutdown here directly?

how about return a VM-Exit related to vcpu shutdown???

Reply:
Yes, I do think it's better to return `SystemDown`. However, we cannot blame the author because `SystemDown` was not presented when `riscv_vcpu` was created. Maybe @PeizhongQiu can changed it in #2? Solving it here may result in conflicts.

Reply:
Ok, I will add this feature in #2.

Reply:
I get that.
sbi related operations should be emulated here, but actual SBI call should only be done in ArceOS.